subprojects/core/src/main/java/org/gradle/api/internal/provider/DefaultProviderFactory.java

```diff
@@ -69,9 +67,9 @@ public DefaultProviderFactory(FileOperations fileOperations) {
             ((PropertyState<Double>) propertyState).set(Double.valueOf(0));
         } else if (clazz == Character.class) {
             ((PropertyState<Character>) propertyState).set(new Character('\0'));
-        } else if (clazz == FileCollection.class || clazz == ConfigurableFileCollection.class) {
+        } else if (clazz == ConfigurableFileCollection.class) {
```

Comment:
So we have 2 different implementations of the same pattern. The next story (or maybe stories) would be to unify these so that they share some stuff and work together well. For example, we might:

- Allow you to switch back and forwards between `FileCollection` and `Provider<Set<File>>` and `ConfigurableFileCollection`  and `PropertyState<Set<File>>` (probably by simply having `FileCollection extends Provider<Set<File>>` and `ConfigurableFileCollection extends PropertySet<Set<File>>`).
- Allow a `Provider<File>` to be added to a file collection and resolved using `project.files()` and all the other places that resolve things to files.
- Allow a file collection to be viewed as a `Provider<File>` in a similar way that you can call `getSingleFile()`.
- Add conveniences for dealing with providers of collection types (which includes file collections) so you can, for example, add elements to the value, etc.